app/functional_tests_fixtures/__init__.py

```diff
@@ -0,0 +1,460 @@
+    environment = current_app.config["NOTIFY_ENVIRONMENT"]
+    test_email_username = os.getenv("TEST_EMAIL_USERNAME", "notify-tests-preview")
+    email_domain = "digital.cabinet-office.gov.uk"
+    function_tests_govuk_key_name = "govuk_notify"
+    function_tests_live_key_name = "functional_tests_service_live_key"
+    function_tests_test_key_name = "functional_tests_service_test_key"
+    govuk_service_id = current_app.config["NOTIFY_SERVICE_ID"]
+    ssm_upload_path = os.getenv("SSM_UPLOAD_PATH")
+
+    org = _create_organiation()
+
+    func_test_user = _create_user(
+        name="Functional Tests",
+        email_address=f"{test_email_username}+{environment}@{email_domain}",
+        password=functional_test_password,
+        mobile_number="07700900001",
+    )
+    email_auth_user = _create_user(
+        name="Functional Tests Email Auth",
+        email_address=f"{test_email_username}+{environment}-email-auth@{email_domain}",
+        password=functional_test_password,
+        auth_type="email_auth",
+    )
+    service_admin_user = _create_user(
+        name="Preview admin tests user",
+        email_address=f"{test_email_username}+{environment}-admin-tests@{email_domain}",
+        password=functional_test_password,
+        organisations=[dao_get_organisation_by_id(org.id)],
+        mobile_number="07700900501",
+    )
+
+    service = _create_service(org.id, service_admin_user)
+
+    dao_add_user_to_service(service, service_admin_user)
+    dao_add_user_to_service(service, email_auth_user)
+
+    _grant_permissions(service, email_auth_user)
+
+    api_key_notify = _create_api_key(function_tests_govuk_key_name, govuk_service_id, service_admin_user.id, "normal")
+    api_key_live_key = _create_api_key(function_tests_live_key_name, service.id, service_admin_user.id, "normal")
+    api_key_test_key = _create_api_key(function_tests_test_key_name, service.id, service_admin_user.id, "test")
+
+    inbound_number_id = _create_inbound_numbers(service.id, service_admin_user.id)
+
+    template1_id = _create_email_template(service, service_admin_user.id)
+    template2_id = _create_sms_template(service, service_admin_user.id)
+    template3_id = _create_letter_template(service, service_admin_user.id)
+
+    _create_service_email_reply_to(
+        service.id, f"{test_email_username}+{environment}-reply-to-default@{email_domain}", True
+    )
+    _create_service_email_reply_to(service.id, f"{test_email_username}+{environment}-reply-to@{email_domain}", False)
+
+    _create_service_permissions(service.id)
+
+    _create_service_sms_senders(service.id, "07700900500", True, inbound_number_id)
+    _create_service_sms_senders(service.id, "func tests", False, None)
+
+    _create_service_inbound_api(service.id, service_admin_user.id)
+
+    functional_test_config = f"""
+
+export FUNCTIONAL_TESTS_API_HOST={current_app.config['API_HOST_NAME']}
+export FUNCTIONAL_TESTS_ADMIN_HOST={current_app.config['ADMIN_BASE_URL']}
```

Comment:
The flow in terms of the concourse pipelines will be the following:

- Terraform will execute this flask task via ecs optionally based on environment. It will run them for the initial install and after each api version update.
- The flask task will write the functional tests config file to SSM. This will change each time as we have generated passwords in there and will be different between environments as guids will change.
- Before the functional tests run they will download the environment file from ssm and source it. The output matches our current functional tests environment file.

Currently the file or ssm are optional based on environment variables passed in. The pipeline will just use ssm. The unit tests in this PR use a file output. I am currently testing it by running the flask command in a different interpreter so can't fake an ssm call. I could try to change that.

I included all the secrets here to just simplify the configuration needed for the functional tests. We could only include environment variables that were generated by the fixtures if you felt that was a better direction to go in.

This [is the pr for the terraform side](https://github.com/alphagov/notifications-aws/pull/2292/files)